<!DOCTYPE html>
<html>
<title>Molecular and Cellular Biology, Oregon State University</title>

<xmp theme="united" style="display:none;">
##Special Topics:  </br>Techniques in Genotyping-by-Sequencing (GBS)</b></br></br>

<b>Course Number: MCB 599 CRN 58928 006: 1 graduate credit P/N</b>
<b><b>Course Meeting Times/Days:</b></b>
<ul>
<li>T/R, 1:30 p.m. - 2:50 p.m., ALS 4000. Spring 2016, April 12 to May 5.</li>
</ul>
</br><b>Course Credits and Grading:  </b>This course combines approximately 20 hours of instruction and assignments for 1 credit. This course will be graded on a Pass/No Pass basis.</br>
</br>
<b>Course Prerequisites:  </b>Graduate standing and/or approval by the instructor. Familiarity with the technology and biological use cases of high throughput sequencing and some experience with command line is required, preferably attendance in MCB 599 Intro to Unix/Command Line. Some basic knowledge or R and command line as well as a basic statistics or previous data analysis course is required in addition to instructor permission.
</br></br>
<b>Instructor: Adelaide Rhodes</b>
<ul>
<li>Office: 3133 ALS; (541)737-1628</li>
<li>Adelaide.Rhodes@cgrb.oregonstate.edu</li>
<li>Scheduled office hours: Tuesday and Thursday, 11:30 a.m. to 12:30 p.m.</li>
<li>Office visits also can be arranged via email.</li>
</ul>
<p><b>Course Webpage:  </b><a href="http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/index.html" class="uri">http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/index.html</a></p>
<p><b>Course Content:</b>This is a lecture/computer lab practicum course to provide an overview of and practical experience in some of the most commonly used methods for GBS analysis. Guest lectures and guided computer exercises will enable participants to apply a variety of methods to the generation and analysis of GBS data. We will also cover variant detection and quality control for SNP calling. Guest lecturers and guided computer exercises will expose participants to practical data management, data quality assessment, visualization tools, and the statistics underlying genotyping-by-sequencing. This class will not cover population genetics in any detail, but will focus more on the preparation of data for downstream analysis from laboratory and field studies.</p>
<p><b>Course Specific Measureable Student Learning Outcomes:</b></p>
<p>At the end of this course, you should be able to:</p>
<ol style="list-style-type: decimal">
<li>Interact more efficiently with core lab personnel on designing and implementing GBS experiments.</li>
<li>Utilize the cloud computing environment at the Center for Genome Research and Biocomputing at Oregon State University.</li>
<li>Run scripts interactively and in batch using command line for GBS Analysis.</li>
<li>Utilize R-scripts for experimental design before GBS and data visualization afterwards.</li>
</ol>
<p><b>Learning Resources:</b><br />
There is no textbook. Course material will be made available as needed on the website or through Blackboard/Canvas throughout the semester.<br />
<br />
<b>Evaluation of Student Performance:</b><br />
Students will be evaluated through computational assignments, quizzes and attendance. Attendance for enrolled students is mandatory. Absences due to reasons other than illness must be approved with the instructor at least 48 hours before the class session.</br></br>
<b>Module I Evaluation Components:</b></p>
<ol style="list-style-type: decimal">
<li>Attendance 80 points (25%)</li>
<li>Homeworks and Quizzes 160 points (50%)</li>
<li>In-class Tutorials 80 points (25%)</li>
</ol>
<p><b>Statement Regarding Students With Disabilities:</b><br />
Accommodations are collaborative efforts between students, faculty and Disability Access Services (DAS). Students with accommodations approved through DAS are responsible for contacting the faculty member in charge of the course prior to or during the first week of the term to discuss accommodations. Students who believe they are eligible for accommodations but who have not yet obtained approval through DAS should contact DAS immediately at 737-4098. DAS website:http://ds.oregonstate.edu<br />
</br>
<b>Academic Honesty</b></br>
The students are expected to be honest and ethical in their academic work. Any incident of academic dishonesty will be handled according to the University's Academic Regulations<a href="http://catalog.oregonstate.edu/ChapterDetail.aspx?key=75#Section2883" class="uri">http://catalog.oregonstate.edu/ChapterDetail.aspx?key=75#Section2883</a><br />
<br />
<b>Link to Statement of Expectations for Student Conduct:</b><br />
<a href="http://studentlife.oregonstate.edu/studentconduct/" class="uri">http://studentlife.oregonstate.edu/studentconduct/</a><br />
<a href="http://oregonstate.edu/main/about/copyright" class="uri">http://oregonstate.edu/main/about/copyright</a><br />
<a href="http://oregonstate.edu/main/about/disclaimer" class="uri">http://oregonstate.edu/main/about/disclaimer</a></p>
<table>
<thead>
<tr class="header">
<th align="left"><b>Week/Date</b></th>
<th align="left"><b>Activity/Location</b></th>
<th align="left"><b>Assignments &amp; Due Dates</b></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">April 5, 2016</td>
<td align="left"><p>Orientation I:</p>
<p>Logging In and File Management</p>
<p><b>ALS 4000</b></p></td>
<td align="left"><p>HW0: Basic Command Line (ungraded)</p>
<p>Due: April 6<sup>th</sup> by 5 p.m., please email directly to Matthew.</p></td>
</tr>
<tr class="even">
<td align="left">April 7, 2016</td>
<td align="left"><p>Orientation II:</p>
<p>More shell commands</p>
<p><b>ALS 4000</b></p></td>
<td align="left"><p><b>PLEASE READ BEFORE CLASS BEGINS ON TUESDAY, APRIL 12<sup>th</sup>.</b></p>
<p>A Robust, Simple Genotyping-by-Sequencing (GBS) Approach for High Diversity Species, Elshire et al. 2011</p>
<p><a href="http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/" class="uri">http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/</a></p>
<p>Genome-wide genetic marker discovery and genotyping using next-generation sequencing. Davey et al. 2011</p>
<p><a href="http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Davey_2011_NatureGenotypingReview.pdf%20">http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Davey_2011_NatureGenotypingReview.pdf</a></p>
<p>GBS Bioinformatics Pipeline ...or, Where Your Data Go After Sequencing Powerpoint Slides <a href="http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Buckler_FilterImpTools111028.pdf" class="uri">http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Buckler_FilterImpTools111028.pdf</a></p></td>
</tr>
<tr class="odd">
<td align="left">April 12, 2016</td>
<td align="left"><p>Syllabus</p>
<p>Intro Lecture</p>
<p>Intro to GBS Javier Tabima</p>
<p>Guest Lecture </p>
<p>Laboratory Methods for GBS Aaron Trippe</p>
<p><b>ALS 4000</b></p></td>
<td align="left"><p>Please finish the above readings before class.</p>
<p>HW1 (20 pts.): Please read through the handout on <em>in silico</em> digest using command line that we will be running as Tutorial 1 on April 14<sup>th</sup>.</p>
<p>Read this paper: Flexible and scalable genotyping-by-sequencing strategies for population studies. Hefelfinger et al. 2014 <a href="http://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-15-979" class="uri">http://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-15-979</a> and answer the following questions. Please email me your responses before noon on April 14<sup>th</sup>.</p>
<ol style="list-style-type: decimal">
<li><p>(4 pts.) What is the difference between a blunt and staggered end restriction enzymes?</p></li>
<li><p>(4 pts.) What are the two organisms they used to test the blunt end enzymes? Name two of the eight blunt end restriction enzymes tested.</p></li>
<li><p>(4 pts.) Why did the Mlyl not retain the appropriate restriction motif as often as the other enzymes chosen?</p></li>
<li><p>(4 pts.) If we had a limited amount of money for sequencing, what fragment size range might be a good choice to achieve near saturation of sites within a limited size spectrum at a lower depth of coverage?</p></li>
<li><p>(4 pts.) Figure 3 discusses how the observed sites matched or did not match the predicted sites from <em>in silico</em> digest. [We are going to be doing <em>in silico</em> digest in class on Thursday].</p></li>
</ol>
<ol style="list-style-type: lower-alpha">
<li><p>How many reads from maize aligned to predicted sites, and</p></li>
<li><p>How many reads from rice aligned to predicted sites? (Hint read down two paragraphs from the figure).</p></li>
</ol></td>
</tr>
<tr class="even">
<td align="left">April 14, 2016</td>
<td align="left"><p>Tutorial 1: <em>In silico</em> digest on command line (10 pts.)</p>
<p>Tutorial 2: Estimating coverage, using R-script (10 pts.)</p>
<p><b>ALS 4000</b></p></td>
<td align="left"><p>HW2 (20 pts.): Repeat Tutorial 2 from In-Class (the R-script) with a larger genome (a link is provided inside the script) and answer the following questions.</p>
<ol style="list-style-type: decimal">
<li><p>5 pts. Discuss whether the individual enzymes and enzyme pair gave a different result than the genome used in class and provide a table comparing the in-class and homework genomes.</p></li>
<li><p>Did the general pattern stay the same  which enzyme would you choose for this project?</p></li>
</ol>
<p>Assigned Reading to be Completed Before April 19:</p>
<p>TASSEL-GBS: A High Capacity Genotyping by Sequencing Analysis Pipeline. Glaubitz et al. 2014</p>
<p><a href="http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Glaubitz_2014_TASSEL.pdf" class="uri">http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Glaubitz_2014_TASSEL.pdf</a></p>
<p>TASSEL: software for association mapping of complex traits</p>
<p>in diverse samples. Bradbury et al. 2007. <a href="http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Bradbury_et_al_2007.pdf" class="uri">http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Bradbury_et_al_2007.pdf</a></p>
<p>Optional Reading:</p>
<p>Structured Association Mapping using STRUCTURE and</p>
<p>TASSEL. Vinod 2011.</p>
<p><a href="http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/TASSEL_STRUCTURE.pdf" class="uri">http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/TASSEL_STRUCTURE.pdf</a></p></td>
</tr>
<tr class="odd">
<td align="left">April 19, 2016</td>
<td align="left"><p>Guest Lecture: Matthew Peterson, CGRB</p>
<p>Tassel Pipeline</p>
<p><b>ALS 4000</b></p></td>
<td align="left"><p>HW3 (20 pts): Read Tutorial 3 (Tassel Pipeline Command Line) and fill in the pipeline diagram (provided as a handout). Turn in this diagram and short answer worksheet at the beginning of the next class (Thursday, April 21 at 1:30 p.m.)</p>
<p>Make sure that the Tassel GUI program is loaded onto your laptop before class on Thursday.</p>
<p>If you are motivated, you can start running the tutorial before class, at least looking at the commands and making sense of them is the first step to running the pipelines on the infrastructure. Please bring your questions to class.</p></td>
</tr>
<tr class="even">
<td align="left">April 21, 2016</td>
<td align="left"><p>Tutorial 3:</p>
<p>Running the Tassel Pipeline (20 pts.)</p>
<p>Interacting with the data on the Tassel GUI interface</p>
<p><b>ALS 4000</b></p></td>
<td align="left"><p>HW 4 (20 pts.): Data analysis on TASSEL program on your computer (20 pts.)</p>
<p>Using the worksheet from the in-class tutorial, answer the questions in the handout. Please turn this in at the beginning of class on April 26<sup>th</sup> at 1:30 p.m.</p>
<p>Assigned Reading to be Completed Before April 26:</p>
<p>Stacks: Building and Genotyping Loci De Novo</p>
<p>From Short-Read Sequences, Catchen et al. 2011</p>
<p><a href="http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Catchen_2011_Stacks_de_Novo.pdf" class="uri">http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Catchen_2011_Stacks_de_Novo.pdf</a></p>
<p>Stacks: an analysis tool set for population genomics, Catchen et al. 2013</p>
<p><a href="http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Catchen_2013_Stacks_population_genomics.pdf" class="uri">http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Catchen_2013_Stacks_population_genomics.pdf</a></p>
<p>GBS-SNP-CROP: a reference-optional pipeline for SNP discovery and plant germplasm characterization using variable length, paired-end genotyping-by-sequencing data Melo et al. 2016. : <a href="http://www.biomedcentral.com/1471-2105/17/29" class="uri">http://www.biomedcentral.com/1471-2105/17/29</a></p></td>
</tr>
<tr class="odd">
<td align="left">April 26, 2016</td>
<td align="left"><p>Guest Lecture:</p>
<p>Kelly Vining</p>
<p>Reference-based GBS Analysis</p>
<p>Javier Tabima</p>
<p>Non-reference based GBS Analysis</p>
<p>Discussion</p>
<p><b>ALS 4000</b></p></td>
<td align="left">HW 5 (20 pts.): Read through the Stacks pipeline handout and fill in the pipeline diagram (provided as a handout). Turn in this diagram and short answer worksheet at the beginning of the next class (Thursday, April 28 at 1:30 p.m.)</td>
</tr>
<tr class="even">
<td align="left">April 28, 2016</td>
<td align="left"><p>Tutorial:</p>
<p>Running the Stacks Pipeline (20 pts.)</p></td>
<td align="left"><p>Assigned Reading to be Completed Before May 3:</p>
<p>Genome-wide association mapping for wood characteristics</p>
<p>in Populus identifies an array of candidate single nucleotide</p>
<p>polymorphisms. Porth et al. 2013</p>
<p><a href="http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Porth_et_al_2013_GWAS.pdf" class="uri">http://teaching.cgrb.oregonstate.edu/MCB/Rhodes/GBS_Spring_2016/Porth_et_al_2013_GWAS.pdf</a></p>
<p>VcfR: an R package to manipulate and visualize VCF format data. Knaus and Grunwald 2016</p>
<p><a href="http://biorxiv.org/content/early/2016/02/26/041277" class="uri">http://biorxiv.org/content/early/2016/02/26/041277</a></p>
<p>HW6 (20 pts.): TBD</p></td>
</tr>
<tr class="odd">
<td align="left">May 3, 2016</td>
<td align="left"><p>Lecture</p>
<p>Quallity Control and Population Genetics</p>
<p>Javier Tabima</p>
<p><b>ALS 4000</b></p></td>
<td align="left">HW7 (20 pts.): TBD</td>
</tr>
<tr class="even">
<td align="left">May 5, 2016</td>
<td align="left">Tutorial 4: VcfR tool for data visualization in R. (20 pts.)</td>
<td align="left">In class quiz  putting it all together (20 pts. for participation)</td>
</tr>
</tbody>
</table>
</xmp>
<script src="http://strapdownjs.com/v/0.2/strapdown.js"></script>
</html>

